ddtrace/contrib/internal/aiokafka/patch.py

```diff
@@ -0,0 +1,201 @@
+from ddtrace.contrib import trace_utils
+from ddtrace.ext import SpanKind
+from ddtrace.ext import SpanTypes
+from ddtrace.ext.kafka import CONSUME
+from ddtrace.ext.kafka import GROUP_ID
+from ddtrace.ext.kafka import HOST_LIST
+from ddtrace.ext.kafka import PRODUCE
+from ddtrace.ext.kafka import SERVICE
+from ddtrace.ext.kafka import TOPIC
+from ddtrace.internal import core
+from ddtrace.internal.constants import COMPONENT
+from ddtrace.internal.constants import MESSAGING_DESTINATION_NAME
+from ddtrace.internal.constants import MESSAGING_SYSTEM
+from ddtrace.internal.schema import schematize_messaging_operation
+from ddtrace.internal.schema import schematize_service_name
+from ddtrace.internal.schema.span_attribute_schema import SpanDirection
+from ddtrace.internal.utils import get_argument_value
+from ddtrace.internal.utils import set_argument_value
+from ddtrace.internal.utils.formats import asbool
+from ddtrace.internal.utils.wrappers import unwrap as _u
+from ddtrace.propagation.http import HTTPPropagator
+
+
+config._add(
+    "aiokafka",
+    dict(
+        _default_service=schematize_service_name("kafka"),
+        distributed_tracing_enabled=asbool(os.getenv("DD_KAFKA_PROPAGATION_ENABLED", default=False)),
+    ),
+)
+
+
+def get_version() -> str:
+    return getattr(aiokafka, "__version__", "")
+
+
+def _supported_versions() -> Dict[str, str]:
+    return {"aiokafka": ">=0.9.0"}
+
+
+def common_aiokafka_tags(topic, bootstrap_servers):
+    return {
+        COMPONENT: config.aiokafka.integration_name,
+        TOPIC: topic,
+        MESSAGING_DESTINATION_NAME: topic,
+        MESSAGING_SYSTEM: SERVICE,
+        HOST_LIST: bootstrap_servers,
+    }
+
+
+def common_consume_aiokafka_tags(topic, bootstrap_servers, group_id):
+    tags = common_aiokafka_tags(topic, bootstrap_servers)
+    tags.update(
+        {
+            SPAN_KIND: SpanKind.CONSUMER,
+            GROUP_ID: group_id,
+        }
+    )
+    return tags
+
+
+def parse_send(instance, args, kwargs):
+    topic = get_argument_value(args, kwargs, 0, "topic")
+    value = get_argument_value(args, kwargs, 1, "value", True)
+    key = get_argument_value(args, kwargs, 2, "key", True) or None
+    partition = get_argument_value(args, kwargs, 3, "partition", True)
+    headers = get_argument_value(args, kwargs, 5, "headers", True) or []
+    servers = instance.client._bootstrap_servers
+
+    return topic, value, headers, partition, key, servers
+
+
+async def traced_send(func, instance, args, kwargs):
+    topic, value, headers, partition, key, bootstrap_servers = parse_send(instance, args, kwargs)
+
+    with core.context_with_data(
+        "aiokafka.send",
+        span_name=schematize_messaging_operation(PRODUCE, provider="kafka", direction=SpanDirection.OUTBOUND),
+        span_type=SpanTypes.WORKER,
+        service=trace_utils.ext_service(None, config.aiokafka),
+        tags=common_aiokafka_tags(topic, bootstrap_servers),
+    ) as ctx:
+        span = ctx.span
```

Comment:
like Emmett mentioned already, we want to avoid any usage of `ctx.span`, especially in new integrations.